templates/vue/src/components/TapestryMain/TapestryNode/index.vue

```diff
@@ -498,6 +498,9 @@ export default {
           ? this.openNode(this.node.id)
           : this.updateRootNode()
       }
+      setTimeout(() => {
+        client.updateUserLastSelectedNode(this.node.id)
+      }, 3000)
```

Comment:
There's a couple of problems here. One is that we are not necessarily updating the selected node here (see the lines before this) so this should be moved to the `updateRootNode()` function instead.

Secondly, part of the requirement was that we do not send the request to save a node if the node was changed within a few seconds. So we should clear this interval if the node is changed. This means we should actually emit an event for this and handle it in the parent component as this is only for this one node. Add a data variable to TapestryMain to keep track of the set timeout and then clear that timeout using the `clearTimeout` function when a node is changed. I would also increase the delay to 5 seconds.

Here's the code just fyi:
```
      clearTimeout(this.updateLastSelectedNodeTimer)
      this.updateLastSelectedNodeTimer = setTimeout(() => {
        client.updateUserLastSelectedNode(this.node.id)
      }, 5000)
```

Reply:
Fixed timeout and saving